pylib/cqlshlib/cqlshmain.py

```diff
@@ -1835,6 +1835,31 @@ def do_help(self, parsed):
                         self.printerr("*** No browser to display CQL help. URL for help topic %s : %s" % (t, url))
             else:
                 self.printerr("*** No help on %s" % (t,))
+    def do_history(self,parsed):
+        """
+        HISTORY [cqlsh only]
+
+           Displays the most recent commands executed in cqlsh
+
+        HISTORY (<n>)
+
+           If n is specified, the history display length is set to n for this session
+        """
+
+        recent_history = ""
+        history_length = readline.get_current_history_length()
+
+        n = parsed.get_binding('n')
+        if (n is not None):
+             val = int(n)
+             self.max_history_length_shown = val
+
+        for index in range(history_length, history_length-self.max_history_length_shown, -1):
+            history_item = readline.get_history_item(index)
+            if history_item:
+                recent_history += history_item +"\n"
+        print("Recent Commands:\n\n" + recent_history)
```

Comment:
I am not sure we need to have "Recent Commands:\n\n". History on *nix does not print it either and it is pretty much self evident that we are printing history here.

Reply:
I agree, I'll remove it